src/git-command-manager.ts

```diff
@@ -0,0 +1,399 @@
+        if (!gitVersion.checkMinimum(minimumGitVersion)) {
+            throw new Error(`Minimum required git version is ${minimumGitVersion}. Your git ('${this.gitPath}') is ${gitVersion}`);
+        }
+
+        if (this.lfs) {
+            // Git-lfs version
+            core.debug("Getting git-lfs version");
+            let gitLfsVersion = new Version();
+            let gitLfsPath = await io.which("git-lfs", true);
+            gitOutput = await this.execGit(["lfs", "version"]);
+            stdout = gitOutput.stdout.trim();
+            if (stdout.indexOf('\n') < 0) {
+                let match = stdout.match(/\d+\.\d+(\.\d+)?/);
+                if (match) {
+                    gitLfsVersion = new Version(match[0]);
+                }
+            }
+            if (!gitLfsVersion.isValid()) {
+                throw new Error("Unable to determine git-lfs version");
+            }
+
+            // Minimum git-lfs version
+            // Note:
+            // - Auth header not supported before 2.1
+            let minimumGitLfsVersion = new Version("2.1");
+            if (!gitLfsVersion.checkMinimum(minimumGitLfsVersion)) {
+                throw new Error(`Minimum required git-lfs version is ${minimumGitLfsVersion}. Your git-lfs ('${gitLfsPath}') is ${gitLfsVersion}`);
+            }
+        }
+
+        // Set the user agent
+        let gitHttpUserAgent = `git/${gitVersion} (github-actions-checkout)`;
+        core.debug(`Set git useragent to: ${gitHttpUserAgent}`);
+        this.gitEnv["GIT_HTTP_USER_AGENT"] = gitHttpUserAgent;
+    }
+
+    private getRandomIntInclusive(
+        minimum: number,
+        maximum: number)
+        : number {
+
+        minimum = Math.floor(minimum);
+        maximum = Math.floor(maximum);
+        return Math.floor(Math.random() * (maximum - minimum + 1)) + minimum;
+    }
+
+    private async sleep(milliseconds): Promise<void> {
+        return new Promise(resolve => setTimeout(resolve, milliseconds));
+    }
+}
+
+class GitOutput {
+    public stdout: string = '';
+    public exitCode: number = 0;
+}
+
+class Version {
+    private readonly major: number = NaN;
+    private readonly minor: number = NaN;
+    private readonly patch: number = NaN;
+
+    constructor(version?: string) {
+        if (version) {
+            let match = version.match(/^(\d+)\.(\d+)(\.(\d+))?$/);
+            if (match) {
+                this.major = Number(match[1]);
+                this.minor = Number(match[2]);
+                if (match[4]) {
+                    this.patch = Number(match[4]);
+                }
+            }
+        }
+    }
+
+    public checkMinimum(minimum: Version): boolean {
+        if (!minimum.isValid()) {
+            throw new Error("Arg minimum is not a valid version");
+        }
+
+        // Major is insufficient
+        if (this.major < minimum.major) {
+            return false;
+        }
+
+        // Major is equal
+        if (this.major == minimum.major) {
```

Comment:
=== (i'll stop adding === comments but you should always use === or !== unless you expect a truthy return value

Reply:
added eslint/prettier